src/Mailgun/Api/Suppressions/Bounce.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/*
+ * Copyright (C) 2013-2016 Mailgun
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+namespace Mailgun\Api;
+
+use Mailgun\Assert;
+use Mailgun\Resource\Api\Suppressions\Bounce\CreateResponse;
+use Mailgun\Resource\Api\Suppressions\Bounce\DeleteResponse;
+use Mailgun\Resource\Api\Suppressions\Bounce\IndexResponse;
+use Mailgun\Resource\Api\Suppressions\Bounce\ShowResponse;
+
+/**
+ * @see https://documentation.mailgun.com/api-suppressions.html#bounces
+ *
+ * @author Sean Johnson <sean@mailgun.com>
+ */
+class Bounce extends HttpApi
+{
+    use Pagination;
+
+    /**
+     * @param string $domain Domain to list bounces for
+     * @param int    $limit  optional
+     *
+     * @return IndexResponse
+     */
+    public function index($domain, $limit = 100)
+    {
+        Assert::string($domain);
+        Assert::notEmpty($domain);
+        Assert::range($limit, 1, 10000, '"Limit" parameter must be between 1 and 10000');
+
+        $params = [
+            'limit' => $limit,
+        ];
+
+        $response = $this->httpGet(sprintf('/v3/%s/bounces', $domain), $params);
+
+        return $this->safeDeserialize($response, IndexResponse::class);
+    }
+
+    /**
+     * @param string $domain  Domain to show bounce from
+     * @param string $address Bounce address to show
+     *
+     * @return ShowResponse
+     */
+    public function show($domain, $address)
+    {
+        Assert::string($domain);
+        Assert::string($address);
+        Assert::notEmpty($domain);
+        Assert::notEmpty($address);
```

Comment:
Sorry, I am feeling like I told you about a waste of time 😅 Glad you agree with this.